src/Scaffolding/VS.Web.CG.Mvc/Minimal Api/MinimalApiGenerator.cs

```diff
@@ -230,32 +231,48 @@ internal static void ValidateModel(MinimalApiGeneratorCommandLineModel model)
             }
         }
 
-        internal async Task ModifyProgramCs(string mapMethodName, string endpointsNamespace)
+        internal async Task ModifyProgramCs(MinimalApiModel templateModel)
         {
+            string endpointsNamespace = templateModel.EndpointsNamespace;
+            string mapMethodName = templateModel.MethodName;
             var jsonText = GetMinimalApiCodeModifierConfig();
             CodeModifierConfig minimalApiChangesConfig = JsonSerializer.Deserialize<CodeModifierConfig>(jsonText);
             if (minimalApiChangesConfig != null)
             {
+                //Getting Program.cs document
                 var programCsFile = minimalApiChangesConfig.Files.FirstOrDefault();
                 programCsFile.Usings = new string[] { endpointsNamespace };
                 var programType = ModelTypesLocator.GetType("<Program>$").FirstOrDefault() ?? ModelTypesLocator.GetType("Program").FirstOrDefault();
                 var project = Workspace.CurrentSolution.Projects.FirstOrDefault(p => p.AssemblyName.Equals(ProjectContext.AssemblyName, StringComparison.OrdinalIgnoreCase));
                 var programDocument = GetUpdatedDocument(project, programType);
-                var docEditor = await DocumentEditor.CreateAsync(programDocument);
 
+                //Modifying Program.cs document
+                var docEditor = await DocumentEditor.CreateAsync(programDocument);
                 var docRoot = docEditor.OriginalRoot as CompilationUnitSyntax;
                 var docBuilder = new DocumentBuilder(docEditor, programCsFile, ConsoleLogger);
                 //adding usings
                 var newRoot = docBuilder.AddUsings(new CodeChangeOptions());
                 //add code snippets/changes.
                 if (programCsFile.Methods != null && programCsFile.Methods.Any())
                 {
-                    var globalChanges = programCsFile.Methods.Where(x => x.Key.Equals("Global", StringComparison.OrdinalIgnoreCase)).First().Value;
-                    foreach (var change in globalChanges.CodeChanges)
+                    var addMethodMapping = programCsFile.Methods.Where(x => x.Key.Equals("Global", StringComparison.OrdinalIgnoreCase)).First().Value;
+                    foreach (var change in addMethodMapping.CodeChanges)
                     {
                         change.Block = string.Format(change.Block, mapMethodName);
                         newRoot = DocumentBuilder.ApplyChangesToMethod(newRoot, new CodeSnippet[] {change}) as CompilationUnitSyntax;
```

Comment:
Possible optimization: if we make "ApplyChangesToMethod" take an IEnumerable<CodeSnippet>, could maybe move line 262 out of the foreach block, and only have to call "ApplyChangesToMethod" once on addMethodMapping.CodeChanges instead of creating a new array for each change

Reply:
so in that particular scenario, there is only one change, so maybe I just need to remove the foreach loop since there is only one iteration. Refactored to make more sense now.

Reply:
Ah gotcha 👍🏻